lib/commons/aria/get-role.js

```diff
@@ -1,10 +1,115 @@
 import getExplicitRole from './get-explicit-role';
 import getImplicitRole from './implicit-role';
+import lookupTable from './lookup-table';
+import isFocusable from '../dom/is-focusable';
 import { getNodeFromTree } from '../../core/utils';
 import AbstractVirtuaNode from '../../core/base/virtual-node/abstract-virtual-node';
 
+// when an element inherits the presentational role from a parent
+// is not defined in the spec, but through testing it seems to be
+// when a specific HTML parent relationship is required and that
+// parent has `role=presentation`, then the child inherits the
+// role (i.e. table, ul, dl). Further testing has shown that
+// intermediate elements (such as divs) break this chain only in
+// Chrome.
+//
+// Also, any nested structure chains reset the role (so two nested
+// lists with the topmost list role=none will not cause the nested
+// list to inherit the role=none).
+//
+// from Scott O'Hara:
+//
+// "the expectation for me, in standard html is that element
+// structures that require specific parent/child relationships,
+// if the parent is set to presentational that should set the
+// children to presentational.  ala, tables and lists."
+// "but outside of those specific constructs, i would not expect
+// role=presentation to do anything to child element roles"
+const inheritsPresentationChain = {
+	// valid parent elements, any other element will prevent any
+	// children from inheriting a presentational role from a valid
+	// ancestor
+	td: ['tr'],
+	th: ['tr'],
+	tr: ['thead', 'tbody', 'tfoot', 'table'],
+	thead: ['table'],
+	tbody: ['table'],
+	tfoot: ['table'],
+	table: [],
+	li: ['ol', 'ul'],
+	ol: [],
+	ul: [],
+	dt: ['dl'],
+	dd: ['dl'],
+	dl: []
+};
+
+// role presentation inheritance.
+// Source: https://www.w3.org/TR/wai-aria-1.1/#conflict_resolution_presentation_none
+function getPresentationalAncestorRole(vNode, explicitRoleOptions) {
```

Comment:
This method did my head in. I had to rewrite it to understand it. Here's my version:

```js
function getInheritedRole(vNode, explicitRoleOptions) {
  const parentNodeNames = inheritsPresentationChain[vNode.props.nodeName];
  if (!parentNodeNames) {
    return null;
  }

  if (!vNode.parent) {
    throw new ReferenceError(
			'Cannot determine role presentational inheritance of a required parent outside the current scope.'
		);
  }

  // parent can't cause inherited presentation
  if (!parentNodeNames.includes(vNode.parent.props.nodeName)) {
    return null
  }

  // Check if parent is presentational:
  const parentRole = getExplicitRole(parent, explicitRoleOptions)
  if (['none', 'presentation'].includes(parentRole) && !hasConflictResolution(parent)) {
    return parentRole
  }

  return getInheritedRole(vNode.parent, explicitRoleOptions)
}
```

Couple improvements:
- it doesn't need the odd empty arrays above
- It doesn't do explicit role lookup on the first node. That lookup, more than anything else made this confusing to me.